addons/website_sale/views/templates.xml

```diff
@@ -1655,7 +1655,9 @@
     <!-- /shop/cart route -->
     <template id="cart" name="Shopping Cart">
         <t t-call="website_sale.checkout_layout">
-            <t t-set="show_shorter_cart_summary" t-value="True"/>
+            <t t-if="website.account_on_checkout != 'mandatory' or not website.is_public_user()">
+                <t t-set="show_shorter_cart_summary" t-value="True"/>
+            </t>
```

Comment:
`show_shorter_cart_summary` is configuration parameter that doesn't show the lines because you are in the cart. In this case it does not make sense to use this one. I'd put the `t-if` on the express_checkout div.